conf/report/report-template.html

```diff
@@ -15,11 +15,267 @@
+        }
+      }
+      else
+        $('#new_options_wrapper_'+current_value).remove();
+      filters.splice(index, 1);
+    }
+
+    function removeAll() {
+      $('#filter_wrapper').remove();
+      iter = 1;
+      var div = $('<div id="filter_wrapper"><button id="mainButton" onclick="addOptions()">Add</button></div>').insertAfter('#headline');
+      filters.length = 0;
+    }
+
+    function applyFilter() {
+      apply_active = true;
+      if(!document.getElementById('removeButton'))
+        var remove_all = $('<button id="removeButton" onclick="removeAll()">Remove all filters</button>').insertAfter('#applyButton');
+      
+      report = $('#report_table').DataTable();
+      var num_filters = filters.length;
+      var coverage, coverage_operator, type, type_operator, tool, tool_operator;
+      for(var i=0; i<num_filters; i++){
+        if(filters[i].mode === 'coverage'){
+          coverage_operator = filters[i].condition;
+          coverage = filters[i].value;
+        }
+        if(filters[i].mode === 'type'){
+          type_operator = filters[i].condition;
+          type = filters[i].value;
+        }
+        if(filters[i].mode === 'tool'){
+          tool_operator = filters[i].condition;
+          tool = filters[i].value;
+        }
+      }
+
+      report.columns().every(function() {
+        var column = this;
+        if(column.index() !== 0 && column.index() !== 1) {
+          var theadname = column.header().textContent.trim();
+
+          if(tool_operator === 'is'){
+            if(theadname === tool){
+              var array = [];
+              coverage = parseInt(coverage);
+              column.data().each(function(d, j) {
+                output = d.split('/');
+                tests_passed = output[0];
+                total_tests = output[1];
+                percentage = parseInt((tests_passed/total_tests)*100);
+                
+                console.log(coverage_operator, coverage)
+                if (coverage_operator === '<'){
+                  if(percentage < coverage){
+                    array.push('^'+d+'$');
+                  }
+                }
+                if (coverage_operator === '<='){
+                  if(percentage <= coverage){
+                    array.push('^'+d+'$');
+                  }
+                }
+                if (coverage_operator === '>'){
+                  if(percentage > coverage){
+                    array.push('^'+d+'$');
+                  }
+                }
+                if (coverage_operator === '>='){
+                  if(percentage >= coverage){
+                    array.push('^'+d+'$');
+                  }
+                }
+                if (coverage_operator === '='){
+                  if(percentage === coverage){
+                    array.push('^'+d+'$');
+                  }
+                }
+              });
+              array = array.join('|');
+              column.search(array, true, false, true).draw();
```

Comment:
> When a filter is active, if we change the value of any dropdown and click on apply, it currently filters from the previous output (i.e, from the dataTable that we get after the first filter is applied)
>  I've tried adding this:
> `report = $('#report_table').DataTable(); report.search('').draw();`
> at the beginning of applyFilters() function to clear any current filter but it didn't seem to make any change

This code applies filter to each column, is this correct? If so, I guess you have to clear search for each column too.